File: blackout-sim/src/dataclasses/producers/coal_plant.py
from datetime import datetime
from ..producer import Producer

class CoalPlant(Producer):
    def __init__(self, id: str = None, created_at: datetime = None) -> None:
        super().__init__(id, created_at)
        self.production_factor = 2

    def get_production(self):
        return super().get_production() * self.production_factor

File: blackout-sim/models/producers/coal_plant.go
package producers

type CoalPlant struct {
	ProductionFactor float64
	BaseProduction   float64
}

func NewCoalPlant() *CoalPlant {
	return &CoalPlant{
		BaseProduction:   100,
		ProductionFactor: 2,
	}
}

func (cp CoalPlant) GetProduction() float64 {
	return cp.BaseProduction * cp.ProductionFactor
}
